src/app/issues-viewer/issues-viewer.component.ts

```diff
@@ -55,10 +62,23 @@ export class IssuesViewerComponent implements OnInit, AfterViewInit, OnDestroy {
     this.groupByChangeSubscription = this.groupingContextService.currGroupBy$.subscribe((newGroupBy) => {
       this.initialize();
     });
+
+    this.popStateEventSubscription = this.router.events
+      .pipe(filter((event) => event instanceof NavigationEnd || event instanceof NavigationStart))
+      .subscribe((event) => {
+        if (event instanceof NavigationStart && event.navigationTrigger === 'popstate') {
+          this.popStateNavigationId = event.id;
+        }
+
+        if (event instanceof NavigationEnd && event.id === this.popStateNavigationId) {
+          this.groupingContextService.initializeFromUrlParams();
+        }
+      });
```

Comment:
Since we already have a ``ViewService::setupFromURL`` function that is only called when a user loads into WATcher via url, why not trigger initialize from URL there?
To keep the url loading functionality in the same place, I think the function should be called there instead of subscribing to the popStateEvent. 
This will also mean we don't arbitrarily load from URL the same groupBy that is already in grouping context whenever we route, instead only loading from URL on initial setup.

Reply:
In my opinion, the ViewService is responsible for handling the current view and current repository. I don't think it would be suitable to set up the grouping service within the ViewService. What if we have a view that doesn't utilize the grouping service?

The reason for subscribing to the popStateEvent is to ensure that when navigating back to the previous issue viewer page, we update our grouping service with the "group by" parameter from the previous page. One scenario is that I set the "group by" value to "Milestone" for the current repository, then I switch to another repository without maintaining the filter. When I click the back button, I should expect to return to the previous repository with the previous "group by" and filters.

Reply:
I see and I am inclined to agree with this behaviour. In that case an excellent use of the pop state event.